src/store/modules/records/mutations.js

```diff
@@ -0,0 +1,87 @@
+import debug from 'debug'
+import Vue from 'vue'
+
+const logger = debug('app:store:records:mutations')
+const logMutation = (mutationName, ...args) => {
+  logger(`${mutationName} mutation being executed`, ...args)
+}
+
+export default {
+  SET_RECORDS(currentState, { listName, records }) {
+    logMutation('SET_RECORDS', listName, records)
+
+    // Filter out any records that don't have metadata attached to them.
+    //  These are records that were created by other providers.
+    currentState[listName] = records.filter((record) => {
+      return !!record.metadata
+    }) || []
+  },
+
+  ADD_RECORD_TO_LIST(currentState, { listName, record }) {
+    logMutation('ADD_RECORD_TO_LIST', listName, record)
+
+    // Filter out any records that don't have metadata attached to them.
+    //  These are records that were created by other providers.
+    if (!record.metadata) {
+      return
+    }
+
+    const exists = currentState[listName].find((existingRecord) => {
+      return existingRecord.tokenId === record.tokenId
+    })
+
+    if (!exists) {
+      currentState[listName].push(record)
+    }
+  },
+
+  REMOVE_RECORD_FROM_LIST(currentState, { listName, record }) {
+    logMutation('REMOVE_RECORD_FROM_LIST', listName, record)
+
+    currentState[listName] = currentState[listName].filter((codexRecord) => {
+      return codexRecord.tokenId !== record.tokenId
+    })
+  },
+
+  SET_ACTIVE_RECORD(currentState, record) {
+    logMutation('SET_ACTIVE_RECORD_INDEX', record)
+
+    currentState.activeRecord = record
+  },
+
+  UPDATE_RECORD_IN_LISTS(currentState, record) {
+    logMutation('UPDATE_RECORD_IN_LISTS', record)
+
+    // Filter out any records that don't have metadata attached to them.
+    //  These are records that were created by other providers.
+    if (!record.metadata) {
+      return
+    }
+
+    const listNames = [
```

Comment:
Hmm, duplicating the list names here instead of reading them dynamically seems like a bad idea.

Maybe something like this?

```javascript
const lists = Object.keys(currentState)
  .filter((key) => {
    return Array.isArray(key)
  })
  // and if you wanted an array of the lists themselves instead of their names...
  .map((key) => {
    return currentState[key]
  })
```

Or like, put all the lists into an object in the state variables like:

```javascript
export default () => {
  return {
    lists: {
      userRecords: [],
      incomingTransfers: [],
      outgoingTransfers: [],
    },
    activeRecord: null,
  }
}
```

Then you could do the above and not worry about the side affects of adding non-record-list-arrays to the state variables (although that would probably work just fine as long as the array never has any null or undefined values.)

I'm just thinking out loud here, feel free to keep this as is since it works just fine.

Reply:
I agree. I like the second recommendation.